pkg/controller/infinispan/infinispan_controller.go

```diff
@@ -760,66 +763,72 @@ func (r *ReconcileInfinispan) statefulSetForInfinispan(m *infinispanv1.Infinispa
-					corev1.ResourceStorage: pvSize,
+		if m.IsDataGrid() {
+			var pvErr error
+			pvSize, pvErr = resource.ParseQuantity(*m.Spec.Service.Container.Storage)
+			if pvErr != nil {
+				return nil, pvErr
+			}
+			if pvSize.Cmp(memory) < 0 {
+				reqLogger.Info("WARNING: persistent volume size is less than memory size. Graceful shutdown may not work.", "Volume Size", pvSize, "Memory", memory)
+			}
+		}
+
+		pvc := &corev1.PersistentVolumeClaim{ObjectMeta: metav1.ObjectMeta{
+			Name:      m.ObjectMeta.Name,
+			Namespace: m.ObjectMeta.Namespace,
+		},
+			Spec: corev1.PersistentVolumeClaimSpec{
+				AccessModes: []corev1.PersistentVolumeAccessMode{
+					"ReadWriteOnce",
+				},
+				Resources: corev1.ResourceRequirements{
+					Requests: corev1.ResourceList{
+						corev1.ResourceStorage: pvSize,
+					},
 				},
 			},
-		},
-	}
-
-	blockOwnerDeletion := false
-	controllerutil.SetControllerReference(m, pvc, r.scheme)
-	pvc.OwnerReferences[0].BlockOwnerDeletion = &blockOwnerDeletion
-	dep.Spec.VolumeClaimTemplates = []corev1.PersistentVolumeClaim{*pvc}
+		}
 
-	// Adding persistent volume mount
-	v := &dep.Spec.Template.Spec.Containers[0].VolumeMounts
-	*v = append(*v, corev1.VolumeMount{
-		Name:      m.ObjectMeta.Name,
-		MountPath: "/opt/infinispan/server/data",
-	})
-	// Adding an init container that run chmod if needed
-	if chmod, ok := os.LookupEnv("MAKE_DATADIR_WRITABLE"); ok && chmod == "true" {
-		initContainerImage := consts.InitContainerImageName
-		dep.Spec.Template.Spec.InitContainers = []corev1.Container{{
-			Image:   initContainerImage,
-			Name:    "chmod-pv",
-			Command: []string{"sh", "-c", "chmod -R g+w /opt/infinispan/server/data"},
-			VolumeMounts: []corev1.VolumeMount{{
-				Name:      m.ObjectMeta.Name,
-				MountPath: "/opt/infinispan/server/data",
-			}},
-		}}
+		blockOwnerDeletion := false
+		controllerutil.SetControllerReference(m, pvc, r.scheme)
+		pvc.OwnerReferences[0].BlockOwnerDeletion = &blockOwnerDeletion
+		dep.Spec.VolumeClaimTemplates = []corev1.PersistentVolumeClaim{*pvc}
+
+		// Adding an init container that run chmod if needed
+		if chmod, ok := os.LookupEnv("MAKE_DATADIR_WRITABLE"); ok && chmod == "true" {
+			initContainerImage := consts.InitContainerImageName
+			dep.Spec.Template.Spec.InitContainers = []corev1.Container{{
+				Image:   initContainerImage,
+				Name:    "chmod-pv",
+				Command: []string{"sh", "-c", "chmod -R g+w /opt/infinispan/server/data"},
+				VolumeMounts: []corev1.VolumeMount{{
+					Name:      m.ObjectMeta.Name,
+					MountPath: "/opt/infinispan/server/data",
+				}},
+			}}
+		}
+	} else {
+		volumes := &dep.Spec.Template.Spec.Volumes
+		ephemeralVolume := corev1.Volume{
+			Name: m.Name,
+			VolumeSource: corev1.VolumeSource{
+				EmptyDir: &corev1.EmptyDirVolumeSource{},
+			},
+		}
+		*volumes = append(*volumes, ephemeralVolume)
+		reqLogger.Info("WARNING: An ephemeral storage is using. All data will be lost after restarting the cluster nodes.")
```

Comment:
" An ephemeral storage is using" doesn't really make sense. How about "WARNING: Ephemeral storage configured. All data will be lost on cluster shutdown and restart."

Reply:
Thanks, fixed